### PR TITLE
Code Quality Fixes

### PR DESCRIPTION
**Pull request by lakshyacis**
core pylint fixes
test_api pylint fixes
test_config pylint fixes

### CODE REVIEW

File: src/tests/test_api.py [hunk 1]
```diff
@@ -31,7 +31,7 @@ class TestAPIShutdown(TestAPIProto, TestProcessShutdown):
     """Separate test case for API command 'shutdown'"""
     def test_shutdown(self):
         """Shutdown the pybitmessage"""
-        self.assertEquals(self.api.shutdown(), 'done')
+        self.assertEqual(self.api.shutdown(), 'done')
```
Comment (g1itch) [inline src/tests/test_api.py]: oops, it's really deprecated since 2.7

File: src/tests/core.py [hunk 2]
```diff
@@ -91,7 +93,7 @@ def test_tcpconnection(self):
             for peer in (state.Peer("127.0.0.1", 8448),):
                 direct = TCPConnection(peer)
                 while asyncore.socket_map:
-                    print("loop, state = %s" % direct.state)
+                    print "loop, state = %s" % direct.state
```
Comment (g1itch) [inline src/tests/core.py]: print function is used intentionally and it is valid for python 2.7.6 (minimum version supported) Any old style prints in code should be replaced if `print` cannot be avoided there (#1373) in order to prepare for python 3 support.

https://docs.python.org/3/whatsnew/3.0.html#print-is-a-function

File: src/tests/core.py [hunk 3]
```diff
@@ -40,6 +41,7 @@ def pickle_knownnodes():
 
 
 def cleanup():
+    """Cleanup knownnodes: removing nodes"""
```
Comment (g1itch) [inline src/tests/core.py]: Cleanup application files

File: src/tests/core.py [hunk 4]
```diff
@@ -25,6 +25,7 @@
 
 
 def pickle_knownnodes():
+    """Organize knownnodes dict"""
```
Comment (g1itch) [inline src/tests/core.py]: Generate old style pickled knownnodes.dat

File: src/tests/test_api.py (hunk 1 above)
Comment (lakshyacis) [reply src/tests/test_api.py]: The source code still in python 2.7, and the method alos deprecates in 2.. So, should i replace with assertEquals or leave it as it is.

File: src/tests/core.py (hunk 2 above)
Comment (lakshyacis) [reply src/tests/core.py]: So should i change it with logger?

File: src/tests/core.py (hunk 3 above)
Comment (lakshyacis) [reply src/tests/core.py]: Okay, thanks!

File: src/tests/core.py (hunk 4 above)
Comment (lakshyacis) [reply src/tests/core.py]: Okay, thanks!

File: src/tests/test_api.py (hunk 1 above)
Comment (g1itch) [reply src/tests/test_api.py]: Yes, replace. I agree (`assertEquals`, which I used is already deprecated), just noticed when you wrote. That's what my comment about. Sorry if it's not clear enough.

File: src/tests/core.py (hunk 2 above)
Comment (g1itch) [reply src/tests/core.py]: Better do not touch this broken test. I'm going to find an appropriate environment to run it. The print was in the original script.

File: src/tests/test_api.py (hunk 1 above)
Comment (lakshyacis) [reply src/tests/test_api.py]: No problem! :)

Comment (lakshyacis) [reply src/tests/test_api.py]: No problem! :)

File: src/tests/core.py (hunk 2 above)
Comment (lakshyacis) [reply src/tests/core.py]: Okay, wil change "print "loop, state = %s" % direct.state" -> print("loop, state = %s" % direct.state).

Comment (PeterSurda) [reply src/tests/core.py]: The print was there for debugging I believe, for tests we can keep it.

**Pull request merged**